bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,263 @@
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, msg + " " + ex.getMessage());
+            return;
+        }
+        this.devicePropertiesCache = Caffeine.newBuilder().maximumSize(10_000)
+                .expireAfterWrite(Duration.ofSeconds(config.getPropertiesRefreshInterval()))
+                .refreshAfterWrite(Duration.ofSeconds(config.getPropertiesRefreshInterval()))
+                .build(this::loadPropertiesForDevice);
+        var scheduledPool = ThreadPoolManager.getScheduledPool("Salus");
+        this.scheduledFuture = scheduledPool.scheduleWithFixedDelay(this::refreshCloudDevices,
+                config.getRefreshInterval() * 2, config.getRefreshInterval(), SECONDS);
+
+        // done
+        updateStatus(ONLINE);
+    }
+
+    private void refreshCloudDevices() {
+        logger.debug("Refreshing devices from CloudBridgeHandler");
+
+        if (!(thing instanceof Bridge bridge)) {
+            logger.debug("No bridge, refresh cancelled");
+            return;
+        }
+        var things = bridge.getThings();
+        for (var thing : things) {
+            if (!thing.isEnabled()) {
+                logger.debug("Thing {} is disabled, refresh cancelled", thing.getUID());
+                continue;
+            }
+            var handler = thing.getHandler();
+            if (handler == null) {
+                logger.debug("No handler for thing {} refresh cancelled", thing.getUID());
+                continue;
+            }
+            var channels = thing.getChannels();
+            for (var channel : channels) {
+                handler.handleCommand(channel.getUID(), REFRESH);
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no commands in this bridge
+        logger.debug("Bridge does not support any commands to any channels. channelUID={}, command={}", channelUID,
+                command);
+    }
+
+    @Override
+    public void dispose() {
+        var localScheduledFuture = scheduledFuture;
+        if (localScheduledFuture != null) {
+            localScheduledFuture.cancel(true);
+            scheduledFuture = null;
+        }
+        super.dispose();
+    }
+
+    @Override
+    public SortedSet<@NonNull DeviceProperty<?>> findPropertiesForDevice(String dsn) {
+        return requireNonNullElse(devicePropertiesCache.get(dsn), emptySortedSet());
+    }
+
+    @Nullable
+    private SortedSet<DeviceProperty<?>> loadPropertiesForDevice(String dsn) {
+        var api = salusApi;
+        if (api == null) {
+            logger.error("Cannot find properties for device {} because salusClient is null", dsn);
+            return null;
+        }
+        logger.debug("Finding properties for device {} using salusClient", dsn);
+        var response = api.findDeviceProperties(dsn);
+        if (response.failed()) {
+            logger.error("Cannot find properties for device {} using salusClient\n{}", dsn, response.error());
```

Comment:
```suggestion
            logger.warn("Cannot find properties for device {} using salusClient\n{}", dsn, response.error());
```

If the proeprty cannot be found, is that a case theat the thing should go offline?

Reply:
I would not like to go down with whole thing cause it can have more properties (channels) that are working